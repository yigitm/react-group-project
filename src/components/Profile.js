import React from 'react';

const Profile = () => {
  return <h1>Under Construction Profile</h1>;
};
export default Profile;
