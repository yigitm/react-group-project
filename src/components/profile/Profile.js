import React from 'react';
import { useSelector } from 'react-redux';
import MyMission from './MyMission';

const Profile = () => {
  const missions = useSelector((state) => state.missionsReducer);

  function getMyMissions(mission) {
    return mission.reserved === true;
  }

  const myMissions = missions.filter(getMyMissions);

  return (
    <div>
      <table className="table table-striped w-50">
        <thead>
          <tr>
            <th className="fs-3 border" scope="col">My Missions</th>
          </tr>
        </thead>
        <tbody>
          {myMissions.map((myMission) => (
            <MyMission key={myMission.mission_id} mission={myMission} />
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Profile;
