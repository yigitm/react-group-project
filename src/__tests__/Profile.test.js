import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Accordion from 'react-bootstrap/Accordion';
import 'bootstrap/dist/css/bootstrap.min.css';
import Profile from '../components/Profile';

describe('Profile.js: component test', () => {
  test('Profile.js: should render items that has `reserved: true` property', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: true,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const rockets = [falcon1, falcon9];

    const filteredReservations = rockets.filter((rocket) =>
      rocket.reserved ? rocket : null,
    );

    const reservedList = filteredReservations.map((rocket) => (
      <Accordion key={rocket.id}>
        <Accordion.Item eventKey={rocket.id}>
          <Accordion.Header>{rocket.rocketName}</Accordion.Header>
          <Accordion.Body>{rocket.description}</Accordion.Body>
        </Accordion.Item>
      </Accordion>
    ));

    const tree = render(
      <Provider store={store}>
        <Container className="mt-5">
          <Row>
            <Col className="col-6 d-flex">
              <h1>My Missions</h1>
            </Col>
            <Col className="col-6 d-flex flex-column">
              <h1>My Rockets</h1>
              {reservedList}
            </Col>
          </Row>
        </Container>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Profile.js: should return items that has `reserved: true` property', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: true,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const rockets = [falcon1, falcon9];

    const filteredReservations = rockets.filter((rocket) =>
      rocket.reserved ? rocket : null,
    );

    expect(filteredReservations.length).toEqual(1);
  });

  test('Profile.js: should NOT return items that has `reserved: false` property', () => {
    const falcon1 = {
      id: '1',
      rocketName: 'Falcon 1',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const falcon9 = {
      id: '2',
      rocketName: 'Falcon 9',
      description: 'Lorem ipsum onum',
      flickrImages: 'image_url',
      reserved: false,
    };
    const rockets = [falcon1];

    const filteredReservations = rockets.filter((rocket) =>
      rocket.reserved ? rocket : null,
    );

    filteredReservations.map((reservation) => {
      expect(reservation.reserved.length).toEqual(0);
    });
  });
});
