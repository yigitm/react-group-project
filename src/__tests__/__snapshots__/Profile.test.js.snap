// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.js: component test Profile.js: should NOT render items that has \`reserved: false\` property 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Profile.js: component test Profile.js: should render items that has \`reserved: true\` property 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5 container"
      >
        <div
          class="row"
        >
          <div
            class="col-6 d-flex col"
          >
            <h1>
              My Missions
            </h1>
          </div>
          <div
            class="col-6 d-flex flex-column col"
          >
            <h1>
              My Rockets
            </h1>
            <div
              class="accordion"
            >
              <div
                class="accordion-item"
              >
                <h2
                  class="accordion-header"
                >
                  <button
                    aria-expanded="false"
                    class="accordion-button collapsed"
                    type="button"
                  >
                    Falcon 1
                  </button>
                </h2>
                <div
                  class="accordion-collapse collapse"
                >
                  <div
                    class="accordion-body"
                  >
                    Lorem ipsum onum
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5 container"
    >
      <div
        class="row"
      >
        <div
          class="col-6 d-flex col"
        >
          <h1>
            My Missions
          </h1>
        </div>
        <div
          class="col-6 d-flex flex-column col"
        >
          <h1>
            My Rockets
          </h1>
          <div
            class="accordion"
          >
            <div
              class="accordion-item"
            >
              <h2
                class="accordion-header"
              >
                <button
                  aria-expanded="false"
                  class="accordion-button collapsed"
                  type="button"
                >
                  Falcon 1
                </button>
              </h2>
              <div
                class="accordion-collapse collapse"
              >
                <div
                  class="accordion-body"
                >
                  Lorem ipsum onum
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
